CMakeLists.txt

```diff
@@ -119,14 +119,17 @@ set(CHI_LIBS stdc++ lua m dl ${MPI_CXX_LIBRARIES} petsc ${VTK_LIBRARIES})
 
 #================================================ Compiler flags
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable")
     set(CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG} -O0")
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable -Wno-c++17-extensions")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable")
     set(CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG} -O0")
 elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable")
-    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable -Wno-sign-compare")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic")
```

Comment:
We need to find a way to not involve pedantic checks for external libraries. With gcc10.2 the compile warnings are going crazy with things like

```
VTK-8.2.0/Common/DataModel/vtkPointData.h:36:50: warning: extra ';' [-Wpedantic]
```

Can we turn it off somehow for certain includes?

Reply:
Yes, I really hate this.  The best way is probably to use -isystem for some includes.  If we have issues with that, we can use individual pragmas, but that's likely to get messy fast.  -isystem should work for gcc and clang.

Reply:
@Naktakala With GCC-11.1.0 and VTK-9.0.1 I get these same warnings.

Reply:
@Naktakala, @zachhardy: Based on the behaviour that you describe, it appears that, in your build environments, the VTK includes are not correctly registered as system headers, which ultimately are exempt from compiler warnings through the mechanism described by @wdhawkins. Ideally, that occurs internally in ```FindVTK``` and it should not necessary for us to reclassify the include directories produced by the execution of the macro. Of course, we can implement a correction if it turns out to be necessary. However, we need to understand the underlying reason(s) for which this behaviour occurs in your build environments, and why it does not occur in mine, in order for me to be able to make an appropriate modification. Otherwise, one of you will need to undertake the task as I am unable to reproduce the behaviour described by you in any of my multiple build environments. Further coordination to follow offline.

Reply:
Here's my guess as what's going on.  By default, gcc (and probably clang) treats headers found in certain directories as system headers.  I haven't had time to look up the list, but I suspect it's locations such as /usr/include, /include, etc. This list is configurable during gcc compilation.  On machines where the headers aren't in system directories and a default gcc install is being used (class01 comes to mind... since I couldn't convince IT to let us manage it ourselves), we'll get these warnings without using -isystem.  This probably also applies to installations on laptops, etc. where prerequisite packages were installed in non-system locations for various reasons.

Reply:
@wdhawkins I think you are right Daryl. `-isystem` seems to have no effect on my laptop.

Reply:
In general, I agree with the assessment. However, that situation describes exactly one of my build environments (system compiler, non-system dependencies) in which the configuration occurs correctly. In that case, the ```-isystem``` flag appears correctly in the compiler commands, which is fundamental to the suppression of the warnings from VTK. Perhaps we are not using ```FindVTK``` in the most robust manner, as something appears to contribute to a change of behaviour in our differing systems and/or environments? Anyway, this is good, as it assists to focus our attention on the input to/output of that macro.